orchagent/intfsorch.cpp

```diff
@@ -508,35 +637,59 @@ bool IntfsOrch::addRouterIntfs(sai_object_id_t vrf_id, Port &port)
         case Port::PHY:
         case Port::LAG:
             attr.value.s32 = SAI_ROUTER_INTERFACE_TYPE_PORT;
+            attrs.push_back(attr);
             break;
         case Port::VLAN:
             attr.value.s32 = SAI_ROUTER_INTERFACE_TYPE_VLAN;
+            attrs.push_back(attr);
+            break;
+        case Port::SUBPORT:
+            attr.value.s32 = SAI_ROUTER_INTERFACE_TYPE_SUB_PORT;
+            attrs.push_back(attr);
             break;
         default:
             SWSS_LOG_ERROR("Unsupported port type: %d", port.m_type);
             break;
     }
-    attrs.push_back(attr);
 
     switch(port.m_type)
     {
         case Port::PHY:
             attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
             attr.value.oid = port.m_port_id;
+            attrs.push_back(attr);
             break;
         case Port::LAG:
             attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
             attr.value.oid = port.m_lag_id;
+            attrs.push_back(attr);
             break;
         case Port::VLAN:
             attr.id = SAI_ROUTER_INTERFACE_ATTR_VLAN_ID;
             attr.value.oid = port.m_vlan_info.vlan_oid;
+            attrs.push_back(attr);
+            break;
+        case Port::SUBPORT:
+            attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
+            attr.value.oid = port.m_parent_port_id;
+            attrs.push_back(attr);
+
+            attr.id = SAI_ROUTER_INTERFACE_ATTR_VLAN_ID;
+            attr.value.oid = port.m_vlan_info.vlan_oid;
```

Comment:
not oid, can you change that?

Reply:
still pending on this one? what is the action item to resolve this one?

Reply:
We can change to the new spec, and get the control plane working. https://github.com/Azure/sonic-swss/pull/969#discussion_r320927112

Reply:
because we cannot create the vlan oid, so I prefer you to change this and make it correct here.

Reply:
You need to move the SAI ref pointer; otherwise the vs test will fail, not passing the sairedis generic validation still follows the old spec that requires a vlan oid

Try: sudo apt install <selected package>#orchagent: :- meta_generic_validation_cc
localadmin@wenni-vm0:~$ exitATTR_VLAN_ID:SAI_ATTR_VALUE_TYPE_OBJECT_ID attributee
logoutnditional and is mandatory but not passed in attr list